### PR TITLE
Succeed even if the trailing slash is omitted

### PR DESCRIPTION
**Pull request by esoergel**
## Summary
https://dimagi-dev.atlassian.net/browse/QA-3148

I'd like to be able to do this with a better URL pattern, but don't think it's possible.  This is at least explicit, the duplication is minimal (it's essentially registering another valid URL pattern), and I think it's worth the trade-off to provide a better user experience.

## Feature Flag
"Enable the v0.6 Case API"

## Product Description
Succeed even if the trailing slash is omitted

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

This is in response to a QA ticket

### Safety story

This feature is unreleased and in QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/api/urls.py [hunk 1]
```diff
@@ -119,7 +120,8 @@ def api_url_patterns():
               ODataFormMetadataView.as_view(), name=ODataFormMetadataView.table_urlname)
     yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/\$metadata$',
               ODataFormMetadataView.as_view(), name=ODataFormMetadataView.urlname)
-    yield url(r'v0.6/case/', include(case_api_urlpatterns))
+    yield url(r'v0.6/case$', case_api)  # We don't want Django to redirect to add a trailing slash
+    yield url(r'v0.6/case/?', include(case_api_urlpatterns))
```
Comment (kaapstorm) [inline corehq/apps/api/urls.py:124]: This should work. It reduces duplication and allows you to drop `case_api_urlpatterns`, but it's not as simple.
```suggestion
    yield url(r'v0.6/case(?:/(?P<case_id>[\w-]+)?/?)?$', case_api, name='case_api')
```

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/api/urls.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/api/urls.py:124]: Alternate that does not match `v0.6/case//`:
```suggestion
    yield url(r'v0\.6/case(?:/(?P<case_id>[\w-]+))?/?$', case_api, name='case_api')
```

**Review by millerdev — APPROVED** [review]

File: corehq/apps/api/urls.py (hunk 1 above)
Comment (kaapstorm) [reply corehq/apps/api/urls.py:124]: Ah! Nice.

Comment (esoergel) [reply corehq/apps/api/urls.py]: Thanks you two, just tried it out (including with tests) and it works like a charm.  The `\.` didn't escape my notice either - nice touch.

**Review by millerdev — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**